### PR TITLE
Report problems with listening to ports and exit

### PR DESCRIPTION
**Pull request by runephilosof-karnovgroup**
Example output:
```
{"level":"warn","error":"listen tcp :8000: bind: address already in use","time":"2023-08-03T10:15:40+02:00","caller":"/home/code/woodpecker/cmd/server/server.go:249","message":"Not able to redirect from http to https"}
{"level":"fatal","error":"listen tcp 127.0.0.1:8443: bind: address already in use","time":"2023-08-03T10:15:40+02:00","caller":"/home/code/woodpecker/cmd/server/server.go:226"}
Process 52495 has exited with status 1
```

### CODE REVIEW

**Comment by runephilosof-karnovgroup** [comment]
The commit is based on the merge-base of main and release/v1.0 to make it easy to also merge it into release/v1.0

**Comment by runephilosof-karnovgroup** [comment]
CI failed due to seemingly unrelated issues
```
Running golangci-lint
golangci-lint run --timeout 10m
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make: *** [Makefile:131: lint] Error 4
```

`golangci-lint run` completed without errors locally.

File: cmd/server/server.go [hunk 1]
```diff
@@ -198,10 +199,16 @@ func run(c *cli.Context) error {
 					NextProtos: []string{"h2", "http/1.1"},
 				},
 			}
-			return serve.ListenAndServeTLS(
+			err = serve.ListenAndServeTLS(
 				c.String("server-cert"),
 				c.String("server-key"),
 			)
+			if !errors.Is(err, http.ErrServerClosed) {
```
[the remark this answers is not in the document]
Comment (anbraten) [reply cmd/server/server.go]: Yes. Just tested it again and we need to check both

File: cmd/server/server.go [hunk 2]
```diff
@@ -198,10 +199,16 @@ func run(c *cli.Context) error {
...
+				log.Fatal().Err(err).Msg("")
+			} else {
+				log.Info().Msgf("Server listening on '%s' serving https", server.Config.Server.PortTLS)
```
[the remark this answers is not in the document]
Comment (anbraten) [reply cmd/server/server.go]: its blocking. removed the else cases

**Review by 6543 — APPROVED** [review]

**Pull request merged**